core/p2p/p2pv2/server.go

```diff
@@ -153,6 +161,12 @@ func (p *P2PServerV2) SendMessageWithResponse(ctx context.Context, msg *p2pPb.Xu
 	percentage := msgOpts.Percentage
 	p.log.Trace("Server SendMessage with response", "logid", msg.GetHeader().GetLogid(),
 		"msgType", msg.GetHeader().GetType(), "checksum", msg.GetHeader().GetDataCheckSum(), "peers", peers)
+
+	matricLabels := prom.Labels{
+		"bcname": msg.GetHeader().GetBcname(),
+		"type":   msg.GetHeader().GetType().String(),
+	}
+	matrics.DefaultServerMetrics.P2PFlowOut.With(matricLabels).Add(float64(proto.Size(msg)))
```

Comment:
Can grcp interceptors be used to reduce duplicate code?  Examples: UnaryServerInterceptor.